### ISSUE TITLE
Enforcer Group import functionality is broken

### ISSUE THREAD

**Issue by davidrosson** [aquasecurity/terraform-provider-aquasec#104]
It is not currently possible to import existing `aquasec_enforcer_groups` resources. Attempting to use `terraform import` results in the attempted lookup of a `<nil>` enforcer group.

### Terraform Version
`v1.0.11`

### Affected Resource(s)
- `aquasec_enforcer_groups`

### Expected Behavior
The `terraform import` command should enable an existing `aquasec_enforcer_groups` resource to be imported to the state.

### Actual Behavior
The provider passes a `<nil>` value to the API when calling `ac.GetEnforcerGroup`. As a result, no Enforcer Group is found and the import fails.

### Steps to Reproduce
1. Manually create an Enforcer Group via the Aqua Console UI
2. Create a corresponding Terraform resource config
3. Run `terraform import aquasec_enforcer_groups.<resource_id> "<group_id>"`
4. Notice that the import fails

### Root Cause
In the function `resourceEnforcerGroupRead` (in the file `aquasec/resource_enforcer_group.go`), the provider is attempting to fetch the name of the Enforcer Group via the following line of code:
```
name := d.Get("group_id").(string)
```
Although this works fine when the function `resourceEnforcerGroupRead` is used to look up the `group_id` when reading the resource configuration for an **_existing_** resource, or when creating a **_new_** resource, it results in `<nil>` when used in the context of a resource **_import_**. This is because the `terraform-plugin-sdk` package's `Importer.State` method does not access the local resource configuration; rather, it is provided an empty resource configuration containing only the specified `id` string (ref: https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.14.0/helper/schema/provider.go#L374-L375).

Therefore, any attempt to execute the command `terraform import aquasec_enforcer_groups.<resource_id> "<group_id>"` results in the following error:
```
$ terraform import aquasec_enforcer_groups.demo1 demo1
[INFO]  Terraform version: 1.0.11
[INFO]  Go runtime version: go1.16.4
[INFO]  CLI args: []string{"/usr/local/bin/terraform", "import", "aquasec_enforcer_groups.demo1", "demo1"}
[INFO]  CLI command args: []string{"import", "aquasec_enforcer_groups.demo1", "demo1"}
...
[INFO]  provider.terraform-provider-aquasec_v0.8.9: 2022/04/22 10:54:46
[ERROR]  error calling ac.GetEnforcerGroup: <nil>: timestamp=2022-04-22T10:54:46.306+0300
╷
│ Error: enforcer group not found:
```
This error message is thrown by the following line of code (which immediately follows the line where `name` is set):
```
r, err := ac.GetEnforcerGroup(name)
```
Since the `group_id` key does not exist in the (mostly empty) resource configuration provided to the `Importer.State` method, the variable name gets set to `<nil>`; therefore, the call to `GetEnforcerGroup` returns no result, since an Enforcer Group with a `<nil>` name does not exist.

### Proposed Solution
Retrieving the resource's ID value during a `terraform import` execution can be accomplished via the `terraform-plugin-sdk` package's method `d.Id()`. Unfortunately, this method call can't simply be used as a "drop in" replacement for `d.Get("group_id").(string)`, because in the case of the creation of a **_new_** resource, the method `d.Id()` returns `<nil>` (since the method would be looking for the resource's ID in the state, which wouldn't exist yet). Therefore, it is necessary to set the value of `name` conditionally, depending on whether or not the key `group_id` exists in the resource configuration being processed:
```
var name string

groupId, ok := d.GetOk("group_id")

if ok {
    name = groupId.(string)
} else {
    name = d.Id()
}
```
The result is that:
- In the context of a "standard" `terraform apply` execution on an **_existing_** resource, or during the creation of a **_new_** resource, the `d.GetOk("group_id")` method returns the resource ID from the local resource configuration.
- In the context of a resource **_import_**,  the `d.Id()` method returns the proper `id` string representing the Enforcer Group Name, which is provided via the `terraform import aquasec_enforcer_groups.<resource_id> "<group_id>"` command (via the passed `group_id` value).

**Comment by semyonmor** [comment]
Hey, @davidrosson Thank you very much for your help with the resolution of the issues that you raised.
Unfortunately due to the limitation of our CI (which will be solved in the next release) we had a problem running tests on the forked repo,
So we used your suggested changes + some small updates from our side and created one [PR](https://github.com/aquasecurity/terraform-provider-aquasec/pull/117) that will cover all issues and PRs opened by you.

**Comment by davidrosson** [comment]
Hi @semyonmor Thanks for taking a look at these suggestions and getting everything merged in so quickly! Since everything has been handled, please feel free to close my Issues/PRs as needed, or let me know if you'd prefer that I closed them.

**Comment by semyonmor** [comment]
Fixed by #117
